tests/cuda/CMakeLists.txt

```diff
@@ -24,15 +45,17 @@ vecmem_add_test( cuda
    "test_cuda_jagged_vector_view_kernels.cuh"
    "${CMAKE_CURRENT_SOURCE_DIR}/../../cuda/src/utils/cuda_error_handling.hpp"
    "${CMAKE_CURRENT_SOURCE_DIR}/../../cuda/src/utils/cuda_error_handling.cpp"
-   LINK_LIBRARIES CUDA::cudart vecmem::core vecmem::cuda GTest::gtest_main
+   LINK_LIBRARIES CUDA::cudart vecmem::core vecmem::cuda vecmem::testing_cuda_main
                   vecmem_testing_common )
 
 # Set up a separate test that would ensure a C++17 standard. But only with
 # CUDA 11+.
 if( "${CUDAToolkit_VERSION}" VERSION_GREATER_EQUAL "11.0" )
    vecmem_add_test( cuda_cxx17
       "test_cuda_memory_resources.cu"
-      LINK_LIBRARIES vecmem::cuda GTest::gtest_main vecmem_testing_common )
+      "${CMAKE_CURRENT_SOURCE_DIR}/../../cuda/src/utils/cuda_error_handling.hpp"
+      "${CMAKE_CURRENT_SOURCE_DIR}/../../cuda/src/utils/cuda_error_handling.cpp"
+      LINK_LIBRARIES CUDA::cudart vecmem::core vecmem::cuda vecmem::testing_cuda_main vecmem_testing_common )
```

Comment:
Why `vecmem::core`? We must only explicitly link against libraries whose public headers we use. It is up to `vecmem::cuda` to ensure that we can use all of its public headers by just linking against that one library. (This is why it publicly depends on `vecmem::core`.)